### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by ruskom**
Dear Phil,

I just reported some bugs we faced with in the last few days. Issue #12 and Issue #13.
I'd like to suggest my fixes to resolve these issues. 
If you accept these changes, may I ask to retrofit them into the .NET 4.5.2 compatible NuGet package? The 0.1.3 version is targets .NET 4.6 only.

Thanks,
Marton Rusko

### CODE REVIEW

File: ModelMetadataExtensions/ConventionalModelMetadataProvider.cs [hunk 1]
```diff
@@ -38,16 +38,19 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
...
             var defaultResourceType = DefaultResourceType;
             var conventionAttribute = conventionType.GetAttributeOnTypeOrAssembly<MetadataConventionsAttribute>();
-            if (conventionAttribute != null && conventionAttribute.ResourceType != null)
+            if (conventionAttribute != null)
             {
-                defaultResourceType = conventionAttribute.ResourceType;
+                if (conventionAttribute.ResourceType != null)
+                {
+                    defaultResourceType = conventionAttribute.ResourceType;
+                }
```
Comment (Haacked) [inline ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: This change is unnecessary. The C# `&&` operator is short-circuited  so if `conventionAttribute` is null, it won't evaluate the right hand side.

File: ModelMetadataExtensions/ConventionalModelMetadataProvider.cs [hunk 2]
```diff
@@ -38,16 +38,19 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
...
             }
             else if (RequireConventionAttribute)
             {
                 return metadataFactory(attributesList);
             }
 
-            ApplyConventionsToValidationAttributes(attributesList, containerType, propertyName, defaultResourceType);
+            ApplyConventionsToValidationAttributes(attributesList, conventionType, propertyName, defaultResourceType);
```
Comment (Haacked) [inline ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: Nice catch!

**Comment by Haacked** [comment]
In general, I prefer a PR per issue fixed. That way we can evaluate each one in isolation. For example, I am happy with one of your fixes, but the other one I still have quesetions on.

File: ModelMetadataExtensions/ConventionalModelMetadataProvider.cs (hunk 1 above)
Comment (ruskom) [inline ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: Yes, I know how the condition evaluated, but please consider that there is an else branch also which cannot be seen on this diff screen... ;)

**Comment by ruskom** [comment]
I am sorry, I will add two pull requests. Thanks.

File: ModelMetadataExtensions/ConventionalModelMetadataProvider.cs (hunk 1 above)
Comment (Haacked) [inline ModelMetadataExtensions/ConventionalModelMetadataProvider.cs]: *smacks forehead* doh! Silly me! Sorry. Nice catch!

**Pull request closed without merging**